chrome_extension/content.js

```diff
@@ -0,0 +1,156 @@
+  const documentBody = document.body.outerHTML;
+  const documentHead = document.head.outerHTML;
+  const page_url = window.location.href;
+
+  sendMessageToBackgroundScript({
+    action: "captureDocumentState",
+    documentBody: documentBody,
+    documentHead: documentHead,
+    elements: elements,
+    url: page_url,
+    timestamp: Date.now(),
+  });
+}
+
+
+/*
+ * Function to handle click events on any element on a web page.
+ * It sends the element details to the background script.
+*/
+function handleElementClick(event) {
+  const element = event.target;
+  const tagName = element.tagName;
+  const { x, y } = elements[element.id] || {};
+  const value = elements[element.id]?.value || "";
+  const attributes = {};
+
+  for (const attr of element.attributes) {
+    attributes[attr.name] = attr.value;
+  }
+
+  sendMessageToBackgroundScript({
+    action: "elementClicked",
+    tagName: tagName,
+    attributes: attributes,
+    x: x,
+    y: y,
+    value: value,
+    url: window.location.href,
+    timestamp: Date.now(),
+  });
+}
+
+
+/*
+ * Function to create a debounced version of a function.
+ * @param {Function} func - The function to be debounced.
+ * @param {number} delay - The delay in milliseconds.
+*/
+function debounce(func, delay) {
+  let timerId;
+  return function (...args) {
+    if (timerId) {
+      clearTimeout(timerId);
+    }
+    timerId = setTimeout(() => {
+      func.apply(this, args);
+      timerId = null;
+    }, delay);
+  };
+}
+
+
+/*
+  * Function to handle input events on any element on a web page.
+  * It sends the element details to the background script.
+  * @param {Event} event - The input event.
+  * @param {Element} element - The element on which the input event occurred.
+*/
+function handleDebouncedInput(event) {
+  const element = event.target;
+  const { x, y } = elements[element.id];
+  const value = elements[element.id].element.value;
+  const tagName = element.tagName;
+  const attributes = {};
+
+  for (const attr of element.attributes) {
+    attributes[attr.name] = attr.value;
+  }
+
+  sendMessageToBackgroundScript({
+    action: "elementInput",
+    tagName: tagName,
+    attributes: attributes,
+    x: x,
+    y: y,
+    value: value,
+    url: window.location.href,
+    timestamp: Date.now(),
+  });
+}
+
+/* Debounce Input Handler */
+const debouncedInputHandler = debounce(handleDebouncedInput, 500);
+
+/* Call Debounce Input Handler */
+function handleElementInput(event) {
+  debouncedInputHandler(event);
+}
+
+
+/*
+ * Add an click and input information of the element.
+*/
+function addElement(element) {
+  const rect = element.getBoundingClientRect();
+  const x = rect.left + window.scrollX;
+  const y = rect.top + window.scrollY;
+  const value = element.value;
+  if (!element.id) {
+    element.id = element.tagName + "_" + x + "_" + y;
+  }
+  elements[element.id] = { element, x, y, value };
+  element.addEventListener("click", handleElementClick);
```

Comment:
I think we also need to handle the `scroll` event in a similar way to the `click` event.

Reply:
[test_file_1.txt](https://github.com/user-attachments/files/16477027/test_file_1.txt)

Reply:
Please re-review and let me know if there any other change you want.

Besides this, I mark this as resolved.